pkg/extensions/sync/sync_test.go

```diff
@@ -973,21 +972,21 @@ func TestSyncBasicAuth(t *testing.T) {
 		})
 
 		Convey("Verify on demand sync with basic auth", func() {
-			registryName := strings.Replace(strings.Replace(srcBaseURL, "http://", "", 1), "https://", "", 1)
+			registryName := sync.StripRegistryTransport(srcBaseURL)
 			credentialsFile := makeCredentialsFile(fmt.Sprintf(`{"%s":{"username": "test", "password": "test"}}`, registryName))
 
 			syncRegistryConfig := sync.RegistryConfig{
-				URL:      srcBaseURL,
+				URLs:     []string{srcBaseURL},
 				OnDemand: true,
 			}
 
 			unreacheableSyncRegistryConfig1 := sync.RegistryConfig{
-				URL:      "localhost:999999",
+				URLs:     []string{"localhost:999999"},
```

Comment:
I think the issue here is that we are using an invalid port instead of unreachable zot server. 
Maybe it is a better idea to use a port < 1024 like "99"